src/Tests/Unit/Scrapers/Pipeline/Mediator/FormErrorDiscovery.test.ts

```
@@ -0,0 +1,429 @@
+  const domDoc = dom.window.document;
+  const domWin = dom.window;
+  return {
+    /**
+     * Execute the callback with JSDOM document/window injected as globals.
+     * Jest runs tests within a file sequentially — globalThis swap is safe.
+     * The finally block guarantees restoration even on callback errors.
+     * @param fn - The evaluate callback.
+     * @param arg - The argument passed to the callback.
+     * @returns Resolved callback result.
+     */
+    evaluate: <T>(fn: (arg: unknown) => T, arg: unknown): Promise<T> => {
```

ESLint

[error] 'arg' is defined but never used.

(no-unused-vars)